dialogue-core/src/main/java/com/palantir/dialogue/core/RetryingChannel.java

```diff
@@ -54,34 +56,55 @@
     public ListenableFuture<Response> execute(Endpoint endpoint, Request request) {
         SettableFuture<Response> future = SettableFuture.create();
 
-        Supplier<ListenableFuture<Response>> callSupplier = () -> delegate.execute(endpoint, request);
-        FutureCallback<Response> retryer = new RetryingCallback<>(callSupplier, future);
-        Futures.addCallback(callSupplier.get(), retryer, DIRECT_EXECUTOR);
+        Function<Integer, ListenableFuture<Response>> callSupplier = attempt -> {
+            return delegate.execute(
+                    endpoint,
+                    Request.builder()
+                            .from(request)
+                            .putHeaderParams(
+                                    "X-B3-TraceId", request.headerParams().get("X-B3-TraceId") + "-attempt-" + attempt)
```

Comment:
What’s going on here?

Reply:
Ah this was some leftover from debugging odd behaviour - removed!